### PR TITLE
testdrive: support runtime variables

### PR DESCRIPTION
**Pull request by sploiselle**
This seems to be the most sane way to handle #14911 and seems like it could have more utility in general.

tl;dr--let actions grow the ability to store something in `State` that can be referred to later. The only downside here is that these values cannot be made accessible to `undo`; better to have support for this with some limitations than not at all.

### Motivation

This PR adds a known-desirable feature.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - there are no user-facing behavior changes

### CODE REVIEW

**Comment by benesch** [comment]
Sean, so sorry, but I tried to email you this response to your issue last night and GitHub never posted it:

image

Soo.. would you mind also removing `fn undo` in this change? I don't think it'll be too bad. Just need to move the `undo` logic for kafka-create-topic and kafka-ingest to a single cleanup action that runs at the start and deletes all topics/schema registry subjects that start with "testdrive-".

**Comment by benesch** [comment]
I think this would also allow us to restructure execution of a testdrive script so that there's only one type of variable. Rather than building all the actions and then executing them, we could build the next action, execute it, build the next action, execute it, updating the one type of variable as we go.

**Comment by sploiselle** [comment]
@benesch For sure! That there was a missing message makes some other comments make more sense.
